### PR TITLE
feat: use the "Update Style" API in the monitoring demo

### LINKED ISSUE #463: [FEAT] Update the monitoring demo to use the new Style API
**Is your feature request related to a problem? Please describe.**
The monitoring demo style edges by adding CSS classes. When the zoom level change, the stroke width doesn't follow the zoom level.
So, the stroke width may appear too large or too small depending on the zoom level.
This is the same for the arrow size.

_`Frequency` use case paths not clearly_ 
TODO query param
https://cdn.statically.io/gh/process-analytics/bpmn-visualization-examples/v0.31.0/demo/monitoring-all-process-instances/index.html

zoom level 1 | zoom level 2
---- | ----
image | image

_`Time` use case paths not clearly_ 
TODO query param
https://cdn.statically.io/gh/process-analytics/bpmn-visualization-examples/v0.31.0/demo/monitoring-all-process-instances/index.html

zoom level 1 | zoom level 2
---- | ----
image | image

**Describe the solution you'd like**
https://github.com/process-analytics/bpmn-visualization-js/issues/1030 will provide a new way to style the BPMN Elements. Comparing to the CSS way, using the API set the value in the SVG DOM elements. So, the size of the stroke won't depend on the zoom level and then, the relative

**Describe alternatives you've considered**
Keep CSS classnames and still have the limitation.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by tbouffard**
Remove the CSS class previously used to highlight the paths, and use the API instead. It provides better rendering results when zooming (proportions are kept).

closes #463 

**Live environment of examples**

https://cdn.statically.io/gh/process-analytics/bpmn-visualization-examples/463_monitoring_demo_use_update-style_api/examples/index.html

### Limitations

The line of the edge finishes at the same point of the top of the arrow, so the arrow seems not very sharp.
To avoid this, we should make the line finish prior to the end of the arrow . mxGraph options exist for this

image

### Screenshots

image

image

### CODE REVIEW

**Review by csouchet — CHANGES_REQUESTED** [review]
The legend for the path is broken. There is no more color :slightly\_frowning\_face:

**Comment by tbouffard** [comment]
@csouchet the path legends is now fixed and the colors are set dynamically using the colors defined in the style object used to use the API.
See https://cdn.statically.io/gh/process-analytics/bpmn-visualization-examples/49651d9/demo/monitoring-all-process-instances/index.html

I am giving a try to a custom mxGraph style definition of sequence flows to see if I can handle the limitations (arrows glitch) directly in the demo. If it works, we will decide if we move that directly in the lib later.

**Comment by tbouffard** [comment]
First improvement in the vite project

image

File: demo/monitoring-all-process-instances/js/execution-data/execution-data.js [hunk 1]
```diff
@@ -198,8 +199,49 @@ class ExecutionData {
             }
         };
         if (pathClass) {
-            data.pathClass = pathClass;
+            data.styleUpdate = buildStyleUpdateOptions(pathClass);
```
Comment (csouchet) [inline demo/monitoring-all-process-instances/js/execution-data/execution-data.js:202]: **suggestion:** 
I thought that the property and the functions for the style of the path were not clear. 
It took me a while to understand what they were referring to 🙁

Comment (tbouffard) [reply demo/monitoring-all-process-instances/js/execution-data/execution-data.js:202]: Let's rename
* styleUpdate --> pathStyle
* buildStyleUpdateOptions --> buildPathStyle
* pathClass variable: find a name

**Review by csouchet — APPROVED** [review]
It's ok with the proposition in the comment https://github.com/process-analytics/bpmn-visualization-examples/pull/486#discussion_r1177986198.

**Pull request merged**